Improve migration guide (#470)
cloudant/nodejs-cloudant: MIGRATION.md

## Other differences
1. Using the `dotenv` package to store credentials in a file is not recommended. See the [external file configuration section](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration) in our API docs for handling this feature in our new library.

> 1.  In `cloudant-node-sdk` all operations are performed from the scope of the client instance and
>     not associated with any sub-scope like the database. There is no need to instantiate a
>     database object to interact with documents - the database name is included as part of
>     document operations. For example, in the case of updating a document you would first call
>     `getDocument` to fetch and then `putDocument` to update. As a result of which the `use`
>     operation also became invalid.

Reviewer: ```suggestion
1.  In `cloudant-node-sdk` all request operation performed through a service client instance. 
    There is no need to instantiate a
    database object to interact with documents - the database name is included in the operation request as parameter.
     For example, in the case of updating a document you would first call
    `getDocument({ db: "dbname",docId: 'FishStew'})` to fetch and then `putDocument({ db: "dbname",docId: 'FishStew'})` to update. As a result of which the `use`
    operation also became obsoleted.
```

Author: I tried to align this phrase with our Java migration guide similar phrase, so I'd like to keep the first 2 sentences as is. However, I like the idea of highlighting the `db` required param.

Wdyt of:
```suggestion
1.  In `cloudant-node-sdk` all operations are performed from the scope of the client instance and
    not associated with any sub-scope like the database. 
    There is no need to instantiate a database object to interact with documents - the database name is included as part of
    document operations. For example, in the case of updating a document you would first call
    `` getDocument({ db: `${dbName}`, docId:`${docId}`}) `` to fetch and then `` putDocument({ db: `${dbName}`, docId:`${docId}`}) `` to update. As a result of which the `use`
    operation also became obsoleted.
```

Reviewer: Okay, if you prefer that way. Just `scope of the client instance` and also `sub-scope like the database` for me just sounds weird. The scope is used mainly with tasks and projects...

Author: I looked for the original suggestion: https://github.com/cloudant/java-cloudant/pull/518#discussion_r571911928

Author: 4a390c6

## Other differences
1. Using the `dotenv` package to store credentials in a file is not recommended. See the [external file configuration section](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration) in our API docs for handling this feature in our new library.
1.  In `cloudant-node-sdk` all operations are performed from the scope of the client instance and
    not associated with any sub-scope like the database. There is no need to instantiate a
    database object to interact with documents - the database name is included as part of
    document operations. For example, in the case of updating a document you would first call

>     `` getDocument({ db: `${dbName}`, docId:`${docId}`}) `` to fetch and then `` putDocument({ 
>     db: `${dbName}`, docId:`${docId}`}) `` to update. As a result of which the `use`

Reviewer: ```suggestion
    `getDocument({ db: `${dbName}`, docId:`${docId}`})` to fetch and then `putDocument({ 
    db: `${dbName}`, docId:`${docId}`})` to update. As a result of which the `use`
```

>     `` getDocument({ db: `${dbName}`, docId:`${docId}`}) `` to fetch and then `` putDocument({ 
>     db: `${dbName}`, docId:`${docId}`}) `` to update. As a result of which the `use`

Reviewer: This is just an alternative format suggestion.
```suggestion
    `getDocument({ db: ${dbName}, docId: ${docId}})` to fetch and then `putDocument({ 
    db: ${dbName}, docId: ${docId}})` to update. As a result of which the `use`
```

Author: 6b34b7f where I removed also the `${..}`s

    operation also became obsoleted.
1. Plugins are not supported, but several of the plugin features exist in the new library e.g. IAM, [automatic retries](https://github.com/IBM/ibm-cloud-sdk-common/#automatic-retries) for failed requests.
1. Error handling is not transferable from `@cloudant/cloudant` to `@ibm-cloud/cloudant`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=node#error-handling) in our API docs.

> 1. Custom HTTP client configurations in `@cloudant/cloudant` are not transferable to

Reviewer 2: Well the Node.js `https.Agent` part _can_ be used, but it has to passed differently - I think that is covered by the linked doc anyway.

Author: Wdyt of 
```suggestion
1. Custom HTTP client configurations in `@cloudant/cloudant` can be set differently in
```

Author: Fixed in 2e65344

   `@ibm-cloud/cloudant`. For more information go to the
   [Configuring the HTTP client section](https://github.com/IBM/ibm-cloud-sdk-common/#configuring-the-http-client)
   in the IBM Cloud SDK Common README.

> 1. Authentication errors occur during service instantiation. For example, the code `const

Reviewer 2: Perhaps this should be prefixed with `If`? It reads like a troubleshooting, but it is in a list of differences. Ditto the following point, perhaps they should both move into a `## Troubleshooting`?

Author: Done in 92c35fe

Reviewer 3: I'm +1 with this change in this and the other migration guides.

Author: I made the same change in the other 2 repos. I'm going to merge them in parallel.
